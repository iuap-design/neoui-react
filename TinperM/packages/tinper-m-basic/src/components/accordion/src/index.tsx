import './Accordion.less'
import { attachPropertiesToComponent } from '@utils/AttachPropertiesToComponent'
import Accordion, { AccordionPanel } from './Accordion'

export type { AccordionProps, AccordionPanelProps } from './iAccordion'

export default attachPropertiesToComponent(Accordion, { Panel: AccordionPanel })
