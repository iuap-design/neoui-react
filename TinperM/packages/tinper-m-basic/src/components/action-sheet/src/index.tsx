import './ActionSheet.less'
import { attachPropertiesToComponent } from '@utils/AttachPropertiesToComponent'
import { showActionSheet } from './ActionSheet'
import ActionSheet from './ActionSheet'
export type { ActionSheetShowHandler, } from './ActionSheet'

export type {
  Action,
  ActionSheetProps,
} from './iActionSheet'

export default attachPropertiesToComponent(ActionSheet, { show: showActionSheet, })
