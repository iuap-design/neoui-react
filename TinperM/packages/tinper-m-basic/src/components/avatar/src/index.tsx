import './Avatar.less'
import { attachPropertiesToComponent } from '@utils/AttachPropertiesToComponent'
import Avatar from './Avatar'
import AvatarGroup from './AvatarGroup'

export type { AvatarProps } from './iAvatar'

export default attachPropertiesToComponent(Avatar, { Group: AvatarGroup, })
