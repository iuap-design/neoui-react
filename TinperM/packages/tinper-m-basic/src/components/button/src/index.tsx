import './Button.less'
import Button from './Button'
export default Button
