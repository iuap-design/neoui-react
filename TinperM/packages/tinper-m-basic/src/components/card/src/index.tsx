import './Card.less'
import Card from './Card'
export type { CardProps } from './iCard'

export default Card
