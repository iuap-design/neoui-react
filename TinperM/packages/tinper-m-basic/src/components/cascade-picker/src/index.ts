import { prompt } from './prompt'
import { CascadePicker } from './CascadePicker'
import { attachPropertiesToComponent } from '@utils/AttachPropertiesToComponent';

export type {
  CascadePickerProps,
  CascadePickerRef,
  CascadePickerOption,
} from './iCascadePicker'

export default attachPropertiesToComponent(CascadePicker, { prompt, })
