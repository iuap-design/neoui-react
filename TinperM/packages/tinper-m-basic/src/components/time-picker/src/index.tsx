import './TimePicker.less'

import TimePicker from './TimePicker'

export default TimePicker


