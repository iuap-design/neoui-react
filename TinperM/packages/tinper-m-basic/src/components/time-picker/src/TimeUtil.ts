import { DateMode } from '@components/date-time-picker/src/iDateTimePicker';
import { format } from 'date-fns';
import { TimeMode } from '@components/time-picker/src/iTimePicker';



export const formatTime = (date: Date, timeMode: TimeMode) => {
  console.log(date, timeMode,  format(
    date,
    timeMode
  ))
  return format(
    date,
    timeMode
  )

}
