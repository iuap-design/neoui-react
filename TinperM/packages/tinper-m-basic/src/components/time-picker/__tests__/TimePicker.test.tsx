/** TimePicker.tsx */
import React from 'react';
import { render, screen, within } from "@testing-library/react";
import { TimePicker } from "@tinper/m";
import userEvent from "@testing-library/user-event";
import { TimePickerProps } from "@components/time-picker/src/iTimePicker";
import { mount } from '@tests/mount'
import { muiPrefix } from '@utils/UpdatePrefixs'
const prefixTimePicker = `${muiPrefix}-time-picker`;

describe('TimePicker Component', () => {
  const timeString = '07:08:19'
  const timeArr = [7, 8, 19]

  const renderTimePicker = (props: Partial<TimePickerProps> = {}) => {
    const onOk = jest.fn()
    const onDismiss = jest.fn()
    const fieldid = 'time-picker-fieldid'
    const { container } = render(<TimePicker fieldid={fieldid} onOk={onOk} onDismiss={onDismiss}  {...props} />)
    const getOptionByLabel = async (precision: string, label: number | string) => within(await screen.findByRole('listbox', { name: new RegExp(precision, 'i') })).getByRole('gridcell', { name: new RegExp(label.toString()) })
    const getOptions = async (precision: string) => within(await screen.findByRole('listbox', { name: new RegExp(precision, 'i') })).getAllByRole('gridcell')

    return {
      container,
      user: userEvent.setup(),
      trigger: screen.getByRole('textbox'),
      getConfirmBtn: () => screen.getByRole('button', { name: /确定/, hidden: true }),
      getCancelBtn: () => screen.getByRole('button', { name: /取消/, hidden: true }),
      onOk,
      onDismiss,
      fieldid,
      getOptionByLabel,
      getOptions
    }
  };
  it('should render correctly', () => {
    const {container} = renderTimePicker({defaultValue: timeString})
    expect(container).toMatchSnapshot()
  });
  it('should render defaultValue, <test prop:: defaultValue>', async () => {
    const { trigger } = renderTimePicker({ defaultValue: timeString })
    expect(trigger).toHaveValue(timeString)
  });
  it('should select defaultValue when trigger', async () => {
    const { user, trigger } = renderTimePicker({ defaultValue: timeString })
    await user.click(trigger)
    const cells = await screen.findAllByRole('gridcell', { selected: true })
    timeArr.forEach((item, i) => {
      expect(cells[i]).toHaveTextContent(item.toString())
    })
  });
  it('should render time default format HH:mm:ss, <test prop:: subuitype>', async () => {
    const { trigger } = renderTimePicker({ defaultValue: timeString })
    expect(trigger).toHaveDisplayValue(/^(\d{1,2}):(\d{1,2}):(\d{1,2})$/)
  });
  it('should render time  format HH:mm', async () => {
    const { trigger } = renderTimePicker({ defaultValue: timeString, subuitype: 'HH:mm' })
    expect(trigger).toHaveDisplayValue(/^(\d{1,2}):(\d{1,2})$/)
  });
  it('should render popTitle, <test prop:: popTitle>', async () => {
    const popTitle = 'popTitle'
    const { user, trigger } = renderTimePicker({ popTitle })
    await user.click(trigger)
    expect(screen.getByText(popTitle)).toBeInTheDocument()
  });
  it('should render placeholder, <test prop:: placeholder>', async () => {
    const placeholder = 'placeholder'
    const { user, trigger } = renderTimePicker({ placeholder })
    await user.click(trigger)
    expect(screen.getByPlaceholderText(placeholder)).toBeInTheDocument()
  });
  it('should render hours interval, <test prop:: hourStep>', async () => {
    const { user, trigger, getOptions } = renderTimePicker({ hourStep: 2 });
    await user.click(trigger)
    const minuteOptions = await getOptions('hour')
    expect(minuteOptions).toHaveLength(12)
  });
  it('should render minute interval, <test prop:: minuteStep>', async () => {
    const { user, trigger, getOptions } = renderTimePicker({ minuteStep: 10 });
    await user.click(trigger)
    const minuteOptions = await getOptions('minute')
    expect(minuteOptions).toHaveLength(6)
  });
  it('should render minute interval, <test prop:: secondStep>', async () => {
    const { user, trigger, getOptions } = renderTimePicker({ secondStep: 3 });
    await user.click(trigger)
    const minuteOptions = await getOptions('second')
    expect(minuteOptions).toHaveLength(20)
  });
  it('should not popup  when disabled, <test prop:: disabled>', async () => {
    const { user, trigger } = renderTimePicker({ disabled: true })
    await user.click(trigger)
    expect(screen.queryAllByRole('gridcell')).toEqual([])
  });
  it('should contain mui clsPrefix default, <test prop:: clsPrefix>', async () => {
    const { user, trigger, container } = renderTimePicker({ value: timeString })
    await user.click(trigger)
    expect(container.firstChild).toHaveClass(/^mui/)
  });
  it('should contain provided clsPrefix', async () => {
    const { user, trigger, container } = renderTimePicker({ value: timeString, clsPrefix: 'test' })
    await user.click(trigger)
    expect(container.firstChild).toHaveClass(/^test/)
  });
  it('should contain fieldid, <test prop:: fieldid>', async () => {
    const { user, trigger, fieldid, getConfirmBtn, getCancelBtn } = renderTimePicker({ value: timeString, clsPrefix: 'test' })
    await user.click(trigger)
    expect(getConfirmBtn()).toHaveAttribute('fieldid', expect.stringContaining(fieldid))
    expect(getCancelBtn()).toHaveAttribute('fieldid', expect.stringContaining(fieldid))
  });
  it('should show 上午/下午, <test prop:: use12Hours>', async () => {
    const { user, trigger } = renderTimePicker({ use12Hours: true });
    await user.click(trigger)
    expect(screen.getByText('上午')).toBeInTheDocument()
    expect(screen.getByText('下午')).toBeInTheDocument()
  });
  it("readOnly test, <test prop:: readOnly>", () => {
    const wrapper = mount(<TimePicker readOnly></TimePicker>);
    wrapper.find(`.${prefixTimePicker}`).simulate('click');
    expect(document.getElementsByClassName('mui-picker')[0]).toBeFalsy();
  });
  it("value test, <test prop:: value>", () => {
    const wrapper = mount(<TimePicker value='14:12:10'></TimePicker>);
    expect(wrapper.find(`.${prefixTimePicker}`).find('input').props().value).toEqual('14:12:10');
    // todo 暂时显示有两个mui-time-picker(?)
    wrapper.find(`.${prefixTimePicker}`).at(1).simulate('click');
    expect(document.getElementsByClassName('mui-picker')[0]).toBeTruthy();
  });
  it("onDismiss && onOk && onClearReturn test, <test prop:: onDismiss>, <test prop:: onOk>, <test prop:: onClearReturn>", () => {
    const onDismiss = jest.fn();
    const onOk = jest.fn();
    const onClearReturn = jest.fn();
    const wrapper1 = mount(<TimePicker onDismiss={onDismiss} value='14:12:10' onOk={onOk} onClearReturn={onClearReturn}></TimePicker>);
    expect(onDismiss).toHaveBeenCalledTimes(0);
    expect(onOk).toHaveBeenCalledTimes(0);
    expect(onClearReturn).toHaveBeenCalledTimes(0);
    wrapper1.find(`.${prefixTimePicker}`).at(1).simulate('click');
    document.getElementsByClassName('mui-picker-footer-buttons')[0].getElementsByClassName('mui-button-primary')[0].click();
    expect(onDismiss).toHaveBeenCalledTimes(0);
    expect(onOk).toHaveBeenCalledTimes(1);
    expect(onClearReturn).toHaveBeenCalledTimes(0);
    wrapper1.find(`.${prefixTimePicker}`).at(1).simulate('click');
    document.getElementsByClassName('mui-picker-footer-buttons')[0].getElementsByClassName('mui-button-default')[0].click();
    expect(onDismiss).toHaveBeenCalledTimes(1);
    expect(onOk).toHaveBeenCalledTimes(1);
    expect(onClearReturn).toHaveBeenCalledTimes(0);
    document.getElementsByClassName('clear-return-button')[0].click();
    expect(onClearReturn).toHaveBeenCalledTimes(1);
  });
  it("format test, <test prop:: format>", () => {
    const wrapper = mount(<TimePicker value='14:12:10' format={val => val + "111"}></TimePicker>);
    expect(wrapper.find('input').props().value.slice(-3)).toEqual('111')
  });
});
