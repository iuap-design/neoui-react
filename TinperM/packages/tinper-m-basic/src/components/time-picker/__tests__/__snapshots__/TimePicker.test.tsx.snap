// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker Component should render correctly 1`] = `
<div>
  <div
    class="mui-time-picker"
  >
    <div
      class="mui-time-picker"
    >
      <div
        class="mui-time-picker-input mui-input mui-input-wrapper-readonly"
        fieldid="time-picker-fieldid_input"
      >
        <div
          class="mui-input-wrap-no-tips"
          fieldid="time-picker-fieldid_input_wrap_no_tips"
        >
          <div
            class="mui-input-content"
            fieldid="time-picker-fieldid_input_content"
          >
            <input
              class="mui-input-box"
              fieldid="time-picker-fieldid"
              placeholder="请选择"
              readonly=""
              type="text"
              value="07:08:19"
            />
            <div
              class="mui-input-clear-box"
              fieldid="time-picker-fieldid_input_clear"
              style="display: none;"
            >
              <svg
                class="mui-input-clear"
                fieldid="time-picker-fieldid_input_clear_icon"
                fill="currentColor"
                height="0.32rem"
                id="arcclose-circle-Fill"
                style="width: 0.36rem; height: 0.36rem;"
                viewBox="0 0 1024 1024"
                width="0.32rem"
              >
                <path
                  class="mui-input-clear"
                  d="M512 939.12576c235.64288 0 426.67008-191.0272 426.67008-426.6752 0-235.648-191.0272-426.6752-426.6752-426.6752-235.648 0-426.6752 191.0272-426.6752 426.6752 0 235.648 191.0272 426.6752 426.6752 426.6752zM396.98944 351.32928l276.62336 276.62336a32.6144 32.6144 0 0 1-46.11584 46.12096l-115.26144-115.2512-115.24608 115.25632a32.6144 32.6144 0 0 1-43.79136 2.11456l-2.32448-2.11456a32.6144 32.6144 0 0 1 0-46.12096l115.24096-115.2512-115.24096-115.25632a32.6144 32.6144 0 0 1-2.11456-43.79136l2.11456-2.3296a32.6144 32.6144 0 0 1 46.11584 0z m230.5024 0.01024a32.6144 32.6144 0 0 1 46.12096 46.11584L596.75648 474.3168a32.6144 32.6144 0 0 1-46.12096-46.11584l76.86144-76.86144z"
                  fieldid="time-picker-fieldid_input_clear_icon"
                  fill="currentColor"
                  height="0.32rem"
                  style="width: 0.36rem; height: 0.36rem;"
                  viewBox="0 0 48 48"
                  width="0.32rem"
                />
              </svg>
            </div>
            <div
              class="mui-input-suffix mui-input-suffix-readonly"
              fieldid="time-picker-fieldid_input_suffix"
            >
              <svg
                fill="currentColor"
                height="0.32rem"
                id="arccalendar"
                style="width: 0.44rem; height: 0.44rem;"
                viewBox="0 0 1024 1024"
                width="0.32rem"
              >
                <path
                  d="M636.136727 186.181818c17.128727 0 30.999273 13.870545 30.999273 31.045818v20.666182h103.470545A113.757091 113.757091 0 0 1 884.363636 351.697455v219.22909a31.045818 31.045818 0 0 1-62.045091 0v-74.472727H201.634909v227.607273c0 28.532364 23.179636 51.665455 51.712 51.665454h517.213091c28.532364 0 51.665455-23.133091 51.665455-51.665454v-31.650909a31.045818 31.045818 0 1 1 62.091636 0v31.650909A113.757091 113.757091 0 0 1 770.606545 837.818182H253.393455A113.757091 113.757091 0 0 1 139.636364 724.061091v-372.363636A113.757091 113.757091 0 0 1 253.393455 237.847273h103.470545v-20.666182a31.045818 31.045818 0 1 1 62.045091 0v20.666182h186.181818v-20.666182c0-17.175273 13.917091-31.045818 31.045818-31.045818z m-279.272727 113.757091H253.393455c-28.532364 0-51.665455 23.179636-51.665455 51.758546l-0.046545 82.711272h620.590545V351.697455c0-28.578909-23.133091-51.758545-51.665455-51.758546h-103.470545v62.091636a31.045818 31.045818 0 1 1-62.045091 0V299.938909H418.909091v62.091636a31.045818 31.045818 0 0 1-62.045091 0V299.938909z"
                  fill="currentColor"
                  height="0.32rem"
                  style="width: 0.44rem; height: 0.44rem;"
                  viewBox="0 0 48 48"
                  width="0.32rem"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
